cmake/macros/macro_find_system_library.cmake

```diff
@@ -26,10 +26,10 @@ MACRO(FIND_SYSTEM_LIBRARY)
   LIST(GET _argn 0 _variable)
   LIST(REMOVE_AT _argn 0 1)
 
-  if("${_variable}" MATCHES "^${_variable}$")
+  IF(NOT DEFINED ${_variable})
     FOREACH(_arg ${_argn})
-      LIST(APPEND CMAKE_REQUIRED_LIBRARIES "-l${_arg}")
-      CHECK_CXX_COMPILER_FLAG("" ${_variable})
+      LIST(APPEND CMAKE_REQUIRED_LIBRARIES "${_arg}")
+      CHECK_CXX_SOURCE_COMPILES("int main(){}" ${_variable})
```

Comment:
Similarly, we can no longer check whether `-l<system library>` is a valid linker flag via `CHECK_CXX_COMPILER_FLAG` but have to rely on `CHECK_CXX_SOURCE_COMPILES` instead.